packages/aws-library/src/aws_library/ec2/_models.py

```diff
@@ -14,45 +14,162 @@
     Field,
     NonNegativeFloat,
     NonNegativeInt,
+    StrictFloat,
+    StrictInt,
     StringConstraints,
     field_validator,
 )
 from pydantic.config import JsonDict
 from types_aiobotocore_ec2.literals import InstanceStateNameType, InstanceTypeType
 
+GenericResourceValueType: TypeAlias = StrictInt | StrictFloat | str
+
 
 class Resources(BaseModel, frozen=True):
     cpus: NonNegativeFloat
     ram: ByteSize
+    generic_resources: Annotated[
+        dict[str, GenericResourceValueType],
+        Field(
+            default_factory=dict,
+            description=(
+                "Arbitrary additional resources (e.g. {'threads': 8}). "
+                "Numeric values are treated as quantities and participate in add/sub/compare."
+            ),
+        ),
+    ] = DEFAULT_FACTORY
 
     @classmethod
     def create_as_empty(cls) -> "Resources":
         return cls(cpus=0, ram=ByteSize(0))
 
     def __ge__(self, other: "Resources") -> bool:
-        return self.cpus >= other.cpus and self.ram >= other.ram
+        """operator for >= comparison
+        if self has greater or equal resources than other, returns True
+        This will return True only if all of the resources in self are greater or equal to other
+
+        Note that generic_resources are compared only if they are numeric
+        Non-numeric generic resources must be equal in both or only defined in self
+        to be considered greater or equal
+        """
+        if self == other:
+            return True
+        return self > other
 
     def __gt__(self, other: "Resources") -> bool:
-        return self.cpus > other.cpus or self.ram > other.ram
+        """operator for > comparison
+        if self has resources greater than other, returns True
+        This will return True only if all of the resources in self are greater than other
+
+        Note that generic_resources are compared only if they are numeric
+        Non-numeric generic resources must only be defined in self
+        to be considered greater
+        """
+        if (self.cpus < other.cpus) or (self.ram < other.ram):
+            return False
+
+        keys = set(self.generic_resources) | set(other.generic_resources)
+        for k in keys:
+            a = self.generic_resources.get(k)
+            b = other.generic_resources.get(k)
+            if a is None:
+                return False
+            if b is None:
+                # a is greater as b is not defined
+                continue
+            if isinstance(a, int | float) and isinstance(b, int | float):
+                if a < b:
+                    return False
+            else:
+                # remaining options is a is str and b is str or mixed types
+                assert isinstance(a, str)  # nosec
```

Comment:
so there is no definition of > for strings but you still allow it? is this ok??

Reply:
there is a definition. please read the docstring